pom.client.xml

```diff
@@ -960,8 +960,10 @@
                 --add-opens com.azure.core/com.azure.core.util.serializer=ALL-UNNAMED
                 --add-opens com.azure.core/com.azure.core.http=ALL-UNNAMED
                 --add-opens com.azure.core/com.azure.core.http.policy=ALL-UNNAMED
+                --add-opens com.azure.core/com.azure.core.http.rest=ALL-UNNAMED
                 --add-opens com.azure.core/com.azure.core.util=ALL-UNNAMED
                 --add-opens com.azure.core/com.azure.core.implementation.http=ALL-UNNAMED
+                --add-opens com.azure.core/com.azure.core.implementation.serializer=ALL-UNNAMED
                 --add-opens com.azure.ai.textanalytics/com.azure.ai.textanalytics=ALL-UNNAMED
                 --add-opens com.azure.data.appconfiguration/com.azure.data.appconfiguration=ALL-UNNAMED
```

Comment:
Not related to the PR but we have a lot of module openings within `pom.client.xml` for module specific tests, should these configurations be moved into their specific module?

Reply:
Initially, I put this here because there were only a handful of them and it was better to just have all these in one place. Now, the list has grown quite big and we may want to consider moving it to its own pom file. That said, we have to configure maven surefire plugin in each of the modules and when we have to make a change, we have to do it in all poms instead of one place in pom.client.xml.